Library/Homebrew/brew.sh

```diff
@@ -381,60 +390,62 @@ then
   fi
 else
   HOMEBREW_PRODUCT="${HOMEBREW_SYSTEM}brew"
-  [[ -n "$HOMEBREW_LINUX" ]] && HOMEBREW_OS_VERSION="$(lsb_release -sd 2>/dev/null)"
+  [[ -n "${HOMEBREW_LINUX}" ]] && HOMEBREW_OS_VERSION="$(lsb_release -sd 2>/dev/null)"
   : "${HOMEBREW_OS_VERSION:=$(uname -r)}"
-  HOMEBREW_OS_USER_AGENT_VERSION="$HOMEBREW_OS_VERSION"
-
-  if [[ -n "$HOMEBREW_FORCE_HOMEBREW_ON_LINUX" && -n "$HOMEBREW_ON_DEBIAN7" ]]
+  HOMEBREW_OS_USER_AGENT_VERSION="${HOMEBREW_OS_VERSION}"
+  # shellcheck disable=SC2154
+  if [[ -n "${HOMEBREW_FORCE_HOMEBREW_ON_LINUX}" && -n "${HOMEBREW_ON_DEBIAN7}" ]]
   then
     # Special version for our debian 7 docker container used to build patchelf and binutils
     HOMEBREW_MINIMUM_CURL_VERSION="7.25.0"
   else
     # Ensure the system Curl is a version that supports modern HTTPS certificates.
     HOMEBREW_MINIMUM_CURL_VERSION="7.41.0"
   fi
-  curl_version_output="$($HOMEBREW_CURL --version 2>/dev/null)"
+  curl_version_output="$(${HOMEBREW_CURL} --version 2>/dev/null)"
   curl_name_and_version="${curl_version_output%% (*}"
-  if [[ $(numeric "${curl_name_and_version##* }") -lt $(numeric "$HOMEBREW_MINIMUM_CURL_VERSION") ]]
+  # shellcheck disable=SC2248
```

Comment:
```suggestion
  # Can't use braces here because .....
  # shellcheck disable=SC2248
```

not sure the reasoning here?

Reply:
add `shellcheck disable=SC2248` to avoid the message below. I don't guess other double quotes are necessary since it was already set up.
```
Your Git executable: $(unset git && type -p ${HOMEBREW_GIT})"
                                            ^-------------^ SC2248: Prefer double quoting even when variables don't contain special characters.


Did you mean: 
Your Git executable: $(unset git && type -p "${HOMEBREW_GIT}")"
```

Reply:
add shellcheck `disable=SC2248` to avoid the message below. I don't guess other double quotes are necessary since it was already set up.

```
Your curl executable: $(type -p ${HOMEBREW_CURL})"
                                ^--------------^ SC2248: Prefer double quoting even when variables don't contain special characters.

Did you mean: 
Your curl executable: $(type -p "${HOMEBREW_CURL}")"
```